src/Backoffice/Tasks/Domain/Actions/FindTaskAction.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+namespace Lightit\Backoffice\Tasks\Domain\Actions;
+
+use Lightit\Backoffice\Tasks\Domain\Models\Task;
+
+class FindTaskAction
+{
+    public function __invoke(int $id): Task
+    {
+        return Task::findOrFail($id);
+    }
```

Comment:
```suggestion
    public function __invoke(Task $task): JsonResponse
    {
        return response()->json(['data' => $task]);
    }
```
Could use route model binding. 

On a different note, controllers should return JSON responses explicitly.

One thing to note is that this endpoint wouldn't work in the way it's programmed. This is because this is your route definition:
```
Route::get("tasks/{task}",FindTaskController::class);
```
This means that the parameter that will be injected into the controller will be `$task`, not `$id`. This means that `$id` would equal `null`.

Reply:
Removed action and left the controller similar to your suggestion 

`class FindTaskController 
{
    public function __invoke(Task $task): JsonResponse
    {
        return response()->json(['data' => $task]);
    }
}`